videoretrieval/models/layers/gatedembeddingmodule.py

```diff
@@ -0,0 +1,58 @@
+"""Implementation of the Gated Embedding Module.
+
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+"""
+
+import tensorflow as tf
+
+class GatedEmbeddingModule(tf.keras.layers.Layer):
+    """An implementation of the gating embeding module.
+
+    The gated embedding module takes one tensor (Z_0) of dimension
+    input_dimension as input and returns a tensor of output_dimension.
+
+    The equations that define the gated embedding module are as follows.
+
+    Z_1 = W_1 * Z_0 + B_1
+    Z_2 = Z_1 element wise multiplication by Sigmoid(W_2 * Z_1 + B_2)
```

Comment:
nit: If this is Z2 = multiply(Z1, Sigmoid), maybe we can rephrase the expression or use np.multiply or something similar?

Reply:
Sure! happy to.